File: 020.ValidParentheses/wyrj/ValidParentheses.c
/* 0ms */
bool isValid(char* s) {
    char stack[1000];
    int index = 0, i;
    for (i = 0; i < strlen(s); i++) {
        switch(s[i]) {
            case '(':
            case '{':
            case '[':
                stack[index++] = s[i];
            break;
            case ')':
                if ('(' != stack[--index]) {
                    return false;
                }
            break;
            case ']':
                if ('[' != stack[--index]) {
                    return false;
                }
            break;
            case '}':
                if ('{' != stack[--index]) {
                    return false;
                }
            break;
        }
    }
    return (0 == index) ? true : false;
}

File: 020.ValidParentheses/wyrj/ValidParentheses.js
/* 132ms */
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    var stack = [];
    for (var i = 0; i < s.length; i++) {
        switch(s[i]) {
            case '(':
            case '{':
            case '[':
                stack.push(s[i]);
            break;
            case ')':
                if ('(' !== stack[stack.length - 1]) {
                    return false;
                }
                stack.pop();
            break;
            case ']':
                if ('[' !== stack[stack.length - 1]) {
                    return false;
                }
                stack.pop();
            break;
            case '}':
                if ('{' !== stack[stack.length - 1]) {
                    return false;
                }
                stack.pop();
            break;
        }
    }
    return (0 === stack.length) ? true : false;
};